chapters/functions.tex

```diff
@@ -2834,6 +2834,7 @@ \subsection{External Objects}\label{external-objects}
   The {\lstinline!constructor!} shall have exactly one output argument in which the constructed instance derived from {\lstinline!ExternalObject!} is returned.
   The arguments to the constructor must not -- directly nor indirectly -- depend on the external object being constructed.
   The constructor shall initialize the object, and must not require any other calls to be made for the initialization to be complete (e.g., from an initial algorithm or initial equation).
+  To indicate failure, the \lstinline!constructor! may return a null pointer as an alternative to invoking functions like \lstinline[language=C]!ModelicaError!.
```

Comment:
```suggestion
  To indicate failure, the \lstinline!constructor! may return a null pointer, which is treated in the same way as a failed \lstinline!assert! in Modelica.
```

Reply:
It was the more popular semantics at the meeting, but we decided to think about it one more round before merging.  We could apply the change, and then restart the discussion.

Maybe this would be slightly better wording?
```suggestion
  To indicate failure, the \lstinline!constructor! may return a null pointer, to be treated in the same way as a failed \lstinline!assert! in Modelica.
```

Reply:
ok, that makes sense